### ISSUE TITLE
[BUG] Incorrect Avg watts displayed on power badge

### ISSUE THREAD

**Issue by mnasman1** [cagnulein/qdomyos-zwift#2843]
**Describe the bug**
The average wattage is displaying incorrectly on the power badge.
**To Reproduce**

**Expected behavior**
This was working but changed when I upgraded to version 952.
**Screenshots**

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**
 - Device: [e.g. iPhone6] iPhone 6S
 - OS: [e.g. iOS8.1]15.8.3
 - Version [e.g. 22]2.18.7
![qz1](https://github.com/user-attachments/assets/eb363b24-cd55-4f30-b8a2-0d6cc6b9fa56)
![Thu Nov 28 09_35_43 2024_power_badge](https://github.com/user-attachments/assets/1924b2d7-7003-4570-948d-613f0b0e957d)
![qz2](https://github.com/user-attachments/assets/a8e75714-d64a-4eae-8037-fc0d660cfd10)
![Wed Nov 27 07_16_11 2024_power_badge](https://github.com/user-attachments/assets/2106d255-905e-40d5-b85e-367f989b0b35)

**Append a debug log**

 Follow this guide https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F
  
  
**Additional context**

**Comment by cagnulein** [comment]
@mnasman1 do you have the average power setting enable in the advanced setting? if so now the average power is used for the chart. that's the reason

**Comment by mnasman1** [comment]
I do not see that setting. All I see under advanced settings dealing with watts is disable wattage from machinery. As you can see in the summary screenshot the average is recording correctly.Matt ***@***.*** Nov 28, 2024, at 9:52 AM, Roberto Viola ***@***.***> wrote:﻿
@mnasman1 do you have the average power setting enable in the advanced setting? if so now the average power is used for the chart. that's the reason

**Comment by cagnulein** [comment]
it's called "Power Average 5 sec." under advanced settings

**Comment by cagnulein** [comment]
do you have the debug log of the issue? because i changed only a line of code and I don't understand which is the issue.

the 2 charts above showing the target line that matches the current wattage. So both target line (black) and the current wattage are wrong?

**Comment by cagnulein** [comment]
hah wait are you talking about the averate watt metric in the badge? i just got it now!

**Comment by mnasman1** [comment]
Yes that is what I was referring to lol.Matt ***@***.*** Nov 28, 2024, at 11:55 AM, Roberto Viola ***@***.***> wrote:﻿
hah wait are you talking about the averate watt metric in the badge? i just got it now!

**Comment by cagnulein** [comment]
sorry i was not getting, i was doing too much things at the same time. Just fixed, i'm sending you a new one!

**Comment by cagnulein** [comment]
Sent!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 28 nov 2024 alle 17:57 mnasman1 ***@***.***> ha
scritto:

**Comment by mnasman1** [comment]
This seems to be fixed now. Thanks Roberto.

**Comment by cagnulein** [comment]
That's great! Thank you for the feedback!

**Comment by cagnulein** [comment]
That's great! Thank you for the feedback!
